app/src/main/java/uz/direction/threadshomework/MainActivity.kt

```diff
@@ -1,18 +1,145 @@
 package uz.direction.threadshomework
 
-import android.os.Bundle
+import android.os.*
 import androidx.appcompat.app.AppCompatActivity
 import uz.direction.threadshomework.databinding.ActivityMainBinding
+import kotlin.concurrent.thread
+import kotlin.random.Random
 
 class MainActivity : AppCompatActivity() {
 
-    lateinit var binding: ActivityMainBinding
+    private lateinit var binding: ActivityMainBinding
+
+    private var firstProgressBar: Thread? = null
+    private var secondProgressBar: Thread? = null
+    private var thirdProgressBar: Thread? = null
+    private var fourthProgressBar: Thread? = null
+
+    private val handler1 = object : Handler(Looper.getMainLooper()) {
+        override fun handleMessage(msg: Message) {
+            val bundle = msg.data
+            val sum = bundle.getInt(R.string.progress.toString())
+            binding.progress1.progress = sum
+        }
+
+    }
+    private val handler2 = object : Handler(Looper.getMainLooper()) {
+        override fun handleMessage(msg: Message) {
+            val bundle = msg.data
+            val sum = bundle.getInt(R.string.progress.toString())
+            binding.progress2.progress = sum
+        }
+
+    }
+    private val handler3 = object : Handler(Looper.getMainLooper()) {
+        override fun handleMessage(msg: Message) {
+            val bundle = msg.data
+            val sum = bundle.getInt(R.string.progress.toString())
+            binding.progress3.progress = sum
+        }
+
+    }
+    private val handler4 = object : Handler(Looper.getMainLooper()) {
+        override fun handleMessage(msg: Message) {
+            val bundle = msg.data
+            val sum = bundle.getInt(R.string.progress.toString())
+            binding.progress4.progress = sum
+        }
+
+    }
+
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
-        setContentView(binding.root)
+        val view = binding.root
+        setContentView(view)
+
+        var isRaceGoing = false
+        binding.btnShuffle.setOnClickListener {
+            if (!isRaceGoing) {
+                firstProgressBar = Thread {
```

Comment:
Too repetitive code, you could create threads with loop and use only one handler. Clean Code